### PR TITLE
Media repo support for content erasure

### PR DESCRIPTION
**Pull request by dbkr**
Don't serve up content from users who have erased their content

### CODE REVIEW

**Comment by hawkowl** [comment]
@dbkr Can you add a changelog entry for this (or a misc)? Instructions are in https://github.com/matrix-org/synapse/blob/develop/CONTRIBUTING.rst#changelog

**Comment by hawkowl** [comment]
This looks fine to me, but do we have any plan as far as sytests or such for data erasure?

**Comment by dbkr** [comment]
I believe @richvdh has a plan for actually erasing the content

File: synapse/rest/media/v1/media_repository.py [hunk 1]
```diff
@@ -188,6 +188,11 @@ def get_local_media(self, request, media_id, name):
             respond_404(request)
             return
 
+        user_erased = yield self.store.is_user_erased(media_info['user_id'])
+        if user_erased:
+            respond_404(request)
+            return
```
Comment (richvdh) [inline synapse/rest/media/v1/media_repository.py]: Logging would be useful, here and in the other places we 404

File: changelog.d/3459.feature [hunk 2]
```diff
@@ -0,0 +1 @@
+Refuse to serve content from users who have erased their content
```
Comment (richvdh) [inline changelog.d/3459.feature]: This is kinda nonsensical: if the content is erased, how can we serve it? 

Also: what does "content"  mean, and what does it mean when it is "from" a user?

**Comment by richvdh** [comment]
> I believe @richvdh has a plan for actually erasing the content
Not really, erasure isn't really on my radar right now. 

Also, that doesn't answer hawkowl's question, which was about sytests. I'd like to see tests too, please.

**Comment by clokep** [comment]
This has bitrotted and needs SyTests. I'm going to close this as inactive, but please rebase and reopen if there's plans to finish up this work!

**Pull request closed without merging**